src/decoder.rs

```diff
@@ -526,6 +533,29 @@ impl<R: ReadBytesExt> Kernel<R> {
         }
         Ok(v)
     }
+
+}
+
+impl<R: ReadBytesExt+ReadSlice> Kernel<R> {
+    /// Read `begin` as the length and return that many raw bytes as slice.
+    /// If length is greater than the given `max_len`, `DecodeError::TooLong`
+    /// is returned instead.
+    pub fn raw_data_slice<'x>(&'x mut self, begin: u8, max_len: usize)
+        -> DecodeResult<&'x [u8]>
+    {
+        let len = try!(self.unsigned(begin));
+        if len > max_len as u64 {
+            return Err(DecodeError::TooLong { max: max_len, actual: len })
+        }
+        // TODO(tailhook) should it deal with eintr? Or should read_slice itself?
```

Comment:
I think it should be handled here. In [`Kernel::raw_data`](https://github.com/twittner/cbor-codec/blob/master/src/decoder.rs#L266-L273) we also handle it.